### DISCUSSION TITLE
How to replace existing Mozc in my Ubuntu with a new new one from sources?

### DISCUSSION THREAD

**Discussion by username30** [google/mozc#523] [Q&A]
Hello and first of all, thank you for your great project!

For quite long time I'm struggling with this issue https://bugs.launchpad.net/ubuntu/+source/mozc/+bug/1931554 myself. From what I see, it was already fixed few months ago in official git source codes. But unfortunately latest version of Mozc available for Ubuntu 21.04 - that I'm using - in it's repositories is 2.26.4220.102 with a version that contain the fix is 2.26.4472.100, so straight
>sudo apt update && sudo apt upgrade

does not help.

I was trying to dig into your official docs for building Mozc in Docker environment https://github.com/google/mozc/blob/master/docs/build_mozc_in_docker.md , but unfortunately my knowledge of Ubuntu is lacking to replace present Mozc system with the new one, and I can't find proper guidelines for it in official docs, neither anywhere in answers on askubuntu for example - with later mostly address Mozc installation for older versions of Ubuntu from official apt repository.

Regarding building instructions, after this command
>sudo docker build --rm -t $USER/mozc_ubuntu20.04 .

I get this output
![image](https://user-images.githubusercontent.com/24947452/134821535-ced9b320-56e6-4dd9-9946-df1160a6a28d.png)

and then after
>sudo docker run --interactive --tty --rm $USER/mozc_ubuntu20.04

I just get some new prompt
![image](https://user-images.githubusercontent.com/24947452/134821559-1f1db0aa-c9ab-448c-b5e7-162219452cf8.png)

I figured out I need to put
>bazel build package --config oss_linux -c opt

there, but then I just get
![image](https://user-images.githubusercontent.com/24947452/134822050-da033713-1268-4768-9b54-706b4866ff5c.png)

again... and with my terribliest apologies, this is where I'm stuck. I mean, I see everything went great so far and now I just have to do something trivial to complete update and just replace something somewhere... But I'm just out of clue what and where.

I've tried to ask this question on [askubuntu](https://askubuntu.com/questions/1364631/how-to-upgrade-mozc-in-my-ubuntu-21-04-to-the-latest-build-from-github), but nobody know what to do there as well.

I would appreciate a lot if anybody could give a bit more detailed steps for building and replacement this system from github sources, without waiting for official Ubuntu repository to be updated.

**Comment by hiroyuki-komatsu** [comment]
Hi,

https://github.com/google/mozc/issues/510 is the corresponding issue to https://bugs.launchpad.net/ubuntu/+source/mozc/+bug/1931554.

So, you might want to update `/usr/lib/ibus-mozc/ibus-engine-mozc` with your build.

Please try to copy `bazel-bin/unix/ibus/ibus_mozc` to `/usr/lib/ibus-mozc/ibus-engine-mozc`.

The steps are something like:
* `sudo mv /usr/lib/ibus-mozc/ibus-engine-mozc /usr/lib/ibus-mozc/ibus-engine-mozc_orig`
* `cp -f bazel-bin/unix/ibus/ibus_mozc /tmp`
* `sudo cp /tmp/ibus_mozc /usr/lib/ibus-mozc/ibus-engine-mozc`
* `sudo chmod 755 /usr/lib/ibus-mozc/ibus-engine-mozc`

**Reply by hiroyuki-komatsu** [reply]
I also added the information about build paths and install paths to the document.

https://github.com/google/mozc/blob/master/docs/build_mozc_in_docker.md#install-paths
